manifests/charts/gateways/istio-egress/templates/deployment.yaml

```diff
@@ -217,6 +217,9 @@ spec:
           - name: {{ $key }}
             value: "{{ $val }}"
           {{- end }}
+          {{- with $gateway.envSources }}
```

Comment:
Can we make this an if/else?

So you can use  `gateway.env` OR `gateway.envSources`

This is really an API improvement and the new way is just categorically better than the old way - I understand we cannot just drop the old approach for compat reasons but we can require people using the new way to do so exclusively, rather than create mix-n-match env configs.

Reply:
Yes, I think this will be a better approach that also addresses my concern.